### ISSUE TITLE
[member-ordering] Member field with type function treated as instance member methods.

### ISSUE THREAD

**Issue by ashubham** [typescript-eslint/typescript-eslint#1053]
**Repro**

```TS
class MyComponent {
  onKeyDown: (evt) => any = (e) => undefined;
  onFocus: () => void = () => {};
  onBlur: () => void = () => {};

  private myField: number;
  constructor() {
  }
} 
```

```JSON
{
  "rules": {
    "@typescript-eslint/member-ordering": "error"
  }
}
```

**Expected Result**

No Error.

**Actual Result**

Member `myField` should be declared before all public instance member definitions.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.3` |
| `@typescript-eslint/parser`        | `2.3.3` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `10.11.0` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
As I mentioned in https://github.com/typescript-eslint/typescript-eslint/issues/417#issuecomment-539356693

This is working as intended.
The use case of purely using a "class property with initial value set to a function expression" as storage space for a function reference is actually an edge case in the real world.

The use case that is supported by this rule is that of the core use case - "class property with initial value set to a function expression" is used as an auto-binding method definition.
